src/java/org/apache/cassandra/db/virtual/ExceptionsTable.java

```diff
@@ -0,0 +1,205 @@
+{
+    /**
+     * Buffer of uncaught exceptions which happened while virtual table was not initialized.
+     */
+    public static final List<ExceptionRow> preInitialisationBuffer = new ArrayList<>();
+
+    public static final String EXCEPTIONS_TABLE_NAME = "uncaught_exceptions";
+    private static final String EXCEPTION_CLASS_COLUMN_NAME = "exception_class";
+    private static final String COUNT_COLUMN_NAME = "count";
+    private static final String LAST_MESSAGE_COLUMN_NAME = "last_message";
+    private static final String LAST_STACKTRACE_COLUMN_NAME = "last_stacktrace";
+    private static final String LAST_OCCURENCE_COLUMN_NAME = "last_occurence";
+
+    // please be sure operations on this structure are thread-safe
+    protected final List<ExceptionRow> buffer;
+
+    ExceptionsTable(String keyspace)
+    {
+        this(keyspace, 1000);
+    }
+
+    ExceptionsTable(String keyspace, int maxSize)
+    {
+        super(TableMetadata.builder(keyspace, EXCEPTIONS_TABLE_NAME)
+                           .comment("View into uncaught exceptions")
+                           .kind(TableMetadata.Kind.VIRTUAL)
+                           .partitioner(new LocalPartitioner(UTF8Type.instance))
+                           .addPartitionKeyColumn(EXCEPTION_CLASS_COLUMN_NAME, UTF8Type.instance)
+                           .addRegularColumn(COUNT_COLUMN_NAME, Int32Type.instance)
+                           .addRegularColumn(LAST_MESSAGE_COLUMN_NAME, UTF8Type.instance)
+                           .addRegularColumn(LAST_STACKTRACE_COLUMN_NAME, ListType.getInstance(UTF8Type.instance, false))
+                           .addRegularColumn(LAST_OCCURENCE_COLUMN_NAME, TimestampType.instance)
+                           .build());
+
+        this.buffer = BoundedLinkedList.create(maxSize);
+    }
+
+    @Override
+    public boolean allowFilteringImplicitly()
+    {
+        return true;
+    }
+
+    /**
+     * @param exceptionClass exception class of uncaught exception
+     * @param message        message of given exception
+     * @param stackTrace     whole stacktrace of given exception
+     * @param occurenceTime  time when given exception ocurred
+     */
+    public void add(String exceptionClass,
+                    String message,
+                    StackTraceElement[] stackTrace,
+                    long occurenceTime)
+    {
+        synchronized (buffer)
+        {
+            ExceptionRow mergeInto = null;
+            for (ExceptionRow row : buffer)
+            {
+                if (row.exceptionClass.equals(exceptionClass))
+                {
+                    mergeInto = row;
+                    break;
+                }
+            }
+
+            if (mergeInto == null)
+            {
+                buffer.add(new ExceptionRow(exceptionClass, message, stackTrace, occurenceTime));
+            }
+            else
+            {
+                mergeInto.count += 1;
```

Comment:
if I am not mistaken then a recent repeated exception info will be evicted if a list is full, even it happened just before an exception which makes the list full..